[collectd 6]: Fix some GCC "-Wextra" warnings in collectd core

eero-t opened the pull request:
Continuation for PR https://github.com/collectd/collectd/pull/4243, fixing more warnings from ticket https://github.com/collectd/collectd/issues/4241.

(Related also to v5 branch PR https://github.com/collectd/collectd/pull/3970.)

octo reviewed (approved):
I really like how the code looks with the new union. We should have done this sooner :)

P.S.: Leaving merging to you, since the PR is still marked "WIP".

eero-t commented:
> I really like how the code looks with the new union. We should have done this sooner :)

Ok, I squashed away the other alternative away.

> P.S.: Leaving merging to you, since the PR is still marked "WIP".

Removed WIP and merged.  I'll work next on the other warnings (+ adding units support to sysman plugin).

@octo in the meanwhile it would be good if you can come up with a nice way of adding extra compiler warning options to `Makefile.am` for building the collectd core files, and maybe also rest of sources for `collectd*` binaries (but not plugins, as those will not pass stricter options).
